src/main/java/com/synopsys/integration/detect/lifecycle/run/step/IntelligentModeStepRunner.java

```diff
@@ -145,19 +145,19 @@ public void runOnline(
 
         stepHelper.runAsGroup("Wait for Results", OperationType.INTERNAL, () -> {
             if (operationRunner.createBlackDuckPostOptions().shouldWaitForResults()) {
-                BlackDuckVersion blackDuckServerVersion = blackDuckRunData.getBlackDuckServerVersion().get();
+                BlackDuckVersion blackDuckServerVersion = blackDuckRunData.getBlackDuckServerVersion();
                 BlackDuckVersion minVersion = new BlackDuckVersion(2022, 10, 0);
                 
```

Comment:
I was actually hoping that the checking of the Black Duck version and the can-I-poll-for-ScanIds? decision would be made during the boot phase, and the result of that decision would get passed down so this class could simply act on the decision, rather than making the decision.